Sources/AsyncHTTPClient/Configuration+BrowserLike.swift

```diff
@@ -0,0 +1,38 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2023 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension HTTPClient.Configuration {
+    /// A ``HTTPClient/Configuration`` that tries to mimick the platform's default or prevalent browser as closely as possible.
+    ///
+    /// Platform's default/prevalent browsers that we're trying to match (these might change over time):
+    ///  - macOS: Safari
+    ///  - iOS: Safari
+    ///  - Android: Google Chrome
+    ///  - Linux (non-Android): Google Chrome
+    ///
+    /// - note: Don't rely on the values in here never changing, they will be adapted to better match the platform's default/prevalent browser.
+    public static var browserLike: HTTPClient.Configuration {
```

Comment:
I'm not convinced about this naming. Browser's have very strict rules about when to reuse a connection concerning cookies and auth headers. AHC does not implement any of those rules. It is questionable if AHC should implement those rules at all, since the use-case is very different.

Reply:
> Well, this `Configuration` aims to match this _as closely as possible_. It doesn't say that it's a perfect match and if we can't/won't implement something then that's okay.

I can see your argument here, but I don't think that it is the right one, given that we throw our arms in the air when it comes to connection trust and reuse. All the security guarantees that users might associate with browser are _completely non existent in AHC_.

Reply:
@fabianfett are you okay with keeping the `browserLike` name here?